File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/const_cc_ge0.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class ConstCCGE0(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(ConstCCGE0.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.PL}

File: golang_vm/uPIMulator/src/device/linker/kernel/instruction/cc/const_cc_ge0.go
package cc

import (
	"errors"
)

type ConstCcGe0 struct {
	condition Condition
}

func (this *ConstCcGe0) Init(condition Condition) {
	conditions := map[Condition]bool{
		PL: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *ConstCcGe0) Condition() Condition {
	return this.condition
}
